File: beecrowd_exercicios_em_Cpp/beginner/average_3.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int average(double  media ,bool exame = false){

    double e;

    if(!exame) cout << fixed << setprecision(1) << "Media: " << media << endl;

    if(media > 6.9 || (exame && media > 4.9)){
        cout << "Aluno aprovado.\n";
        if(exame) cout << fixed << setprecision(1) << "Media final: " << media << endl;
        return 0;
    }

    else if (media < 5.0 || (exame && media < 5.0)){
        cout << "Aluno reprovado.\n";
        if(exame) cout << fixed << setprecision(1) << "Media final: " << media << endl;
        return -1;
    }

    cout << "Aluno em exame." << endl;

    cin >> e;

    cout << "Nota do exame: " << e << endl;

    average(media = (media + e)/2,exame = true);

}

int main(){

    double a,b,c,d;

    cin >> a >> b >> c >> d;

    double media = (a*2 + b*3 + c*4 + d*1)/10;

    average(media);

    return 0;
}

File: beecrowd_exercicios_em_python/begginner/average_3.py
def in_exam_student(y):
    print("Aluno em exame.")

    e = float(input())
    print("Nota do exame: {:.1f}".format(e))

    y += e
    y /= 2

    print(check_rslt(y, 0))

    return "Media final: " + str(y)


def check_rslt(x, y=1):
    if x >= 7.0 and y or x >= 5.0 and not y:
        return "Aluno aprovado."

    return "Aluno reprovado."


def get_avg():
    a, b, c, d = map(float, input().split())

    avg = (a * 2 + b * 3 + c * 4 + d * 1) / 10

    return avg


def take_exams():
    rslt = get_avg()

    print("Media: {:.1f}".format(rslt))

    if rslt >= 7.0 or rslt < 5.0:
        return check_rslt(rslt)

    return in_exam_student(rslt)


print(take_exams())
